### ISSUE TITLE
Trying to install version 3.3.1 error 500

### ISSUE THREAD

**Issue by Alex-JTI** [postfixadmin/postfixadmin#429]
I have version 3.2 installed running fine. I've installed version 3.3.1 in the neighboring folder and copied config.local.php to the root of new folder.
Permissions are correct on the new folder machine web server user and I'm able to open /postfixadmin/index.php for test.
But when trying to open /postfixadmin/public/index.php /postfixadmin/public/setup.php or /postfixadmin/public/login.php
it generates 500 error.
There is nothing specific in the web server logs that could explain the issue , so it points to PHP root cause.

When I enabled PHP display errors it is showing up PHP error:
`Parse error: syntax error, unexpected ':', expecting ';' or '{' in /home/ipnets/public_html/pa-3.3/model/Config.php on line 48`

Indeed, source line 48 refers to:
`public static function read_array(string $var) : array {`

Is there any PHP incompatibility with PHP 5.6.37?
Please help to fix the issue.

**Comment by DavidGoodwin** [comment]
Yes indeed. PostfixAdmin 3.3.x requires PHP 7.0 or greater.

**Comment by Alex-JTI** [comment]
Thinks for pointing to this. I've never seen any notice about in neither in install.txt nor in changelog.txt

Now I have installed PHP 7.3 - public/login.php is loaded successfully but shows "Access denied."

Which access it refers to?
BTW I've read in changelog for 3.3:
   - Change setup.php to use PHP's password_hash() for the config setup_password . (breaking change, existing setup passwords will fail to work and need regenerating) 
Does it mean that 
`$CONF['setup_password'] = '61e3df93b9e994e15af186bdb40c97d2:060ebce6f85b878e03f36dfc2ad5b65cb63c1f2e';`
I have in the config file is not valid anymore and breaking the access to SQL?

Is there any way to reset it (setup.php is also not loaded showing the same error)?

**Comment by DavidGoodwin** [comment]
To update your setup_password, visit setup.php and it should allow you to generate a new one. Alternatively use : 
```
php -r "echo password_hash('somegoodpassword', PASSWORD_DEFAULT);"
```

**Comment by DavidGoodwin** [comment]
I'm not sure why you'd be seeing an 'Access denied' message on public/login. That doesn't look like an error Postfixadmin has produced. 

Sorry about not making the PHP version requirement more obvious; see ecf6e4afe9d7fa4c51ba90c56a98d4c5aa070e9f which might help future people.

**Comment by Alex-JTI** [comment]
public/setup.php generates 'Access denied' as well.
And it looks like postfixadmin generated looking at the syntax and . in the end. In the server logs I see 403 error but it doesn't look like apache response, may be it's HP-FPM response since I'm proyxing apache to PHP-FPM 7.3.

I've also generated new hash with `php -r "echo password_hash('somegoodpassword', PASSWORD_DEFAULT);"` and put it in config.local.php but still get the same error message.

**Comment by Alex-JTI** [comment]
Now I'm facing another issue: after putting root credentials I get:
`Invalid token (session timeout; refresh the page and try again?)`
I have generated new hash in setup.php and inserted it to config.local.php but the issue is still there.

**Comment by DavidGoodwin** [comment]
That happens if the session state isn't being carried between one page and another.

Possible causes : 

 * Your browser is blocking or not sending cookies to Postfixadmin - look in the network tab of your browser and see what headers it's sending. The session identifier (in a 'Cookie' header should be stable between reloads of the same page).
 * Your server is unable to store PHP session files (an therefore not able to maintain state across sessions). Check your PHP configuration and look for 'session.save_path' and make sure the webserver can read/write to that location. You should see other files there already, which should be owned by the webserver. Normally a refresh of a Postfixadmin page would lead to at least one file being created in that directory.

**Comment by Alex-JTI** [comment]
@DavidGoodwin 
many thanks for your support! Indeed it was web server not able to write to path in 'session.save_path'

**Comment by Alex-JTI** [comment]
I'm able to login to the panel and open some pages like /public/main.php and public/viewlog.php but when navigating to mailbox and domain sections like /public/list.php?table=domain I get 500 error.

I see the following in the error log:
```
[Fri Jan 22 00:56:45.424237 2021] [proxy_fcgi:error] [pid 25476:tid 139733908829952] [client 93.92.199.232:52640] AH01071: Got error 'PHP message: Config::read(): attempt to read undefined config option "database_socket", returning nullPHP message: PHP Notice: Undefined index: theme_favicon in /home/ipnets/public_html/postfixadmin/model/PFASmarty.php on line 73PHP message: PHP Notice: Undefined index: show_vacation in /home/ipnets/public_html/postfixadmin/functions.inc.php on line 2179PHP message: PHP Notice: Undefined index: show_disabled in /home/ipnets/public_html/postfixadmin/functions.inc.php on line 2189PHP message: Config::read(): attempt to read undefined config option "password_expiration", returning nullPHP message: PHP Fatal error: In bool: expected config password_expiration to be a string, but received a NULL in /home/ipnets/public_html/postfixadmin/model/Config.php on line 140', referer: https://pa.ipnets.ru/public/main.php

[Fri Jan 22 01:05:34.160793 2021] [proxy_fcgi:error] [pid 15559:tid 139733866866432] [client 93.92.199.232:52866] AH01071: Got error 'PHP message: PHP Notice: Undefined index: theme_favicon in /home/ipnets/public_html/postfixadmin/model/PFASmarty.php on line 73', referer: https://pa.ipnets.ru/public/login.php

[Fri Jan 22 01:05:54.612471 2021] [proxy_fcgi:error] [pid 15559:tid 139733791332096] [client 93.92.199.232:52878] AH01071: Got error 'PHP message: PHP Notice: Undefined index: theme_favicon in /home/ipnets/public_html/postfixadmin/model/PFASmarty.php on line 73PHP message: Config::read(): attempt to read undefined config option "database_socket", returning null', referer: https://pa.ipnets.ru/public/main.php

[Fri Jan 22 01:06:10.813141 2021] [proxy_fcgi:error] [pid 15559:tid 139733934008064] [client 93.92.199.232:52888] AH01071: Got error 'PHP message: PHP Notice: Undefined index: theme_favicon in /home/ipnets/public_html/postfixadmin/model/PFASmarty.php on line 73PHP message: Config::read(): attempt to read undefined config option "database_socket", returning nullPHP message: Config::read(): attempt to read undefined config option "password_expiration", returning nullPHP message: PHP Fatal error: In bool: expected config password_expiration to be a string, but received a NULL in /home/ipnets/public_html/postfixadmin/model/Config.php on line 140', referer: https://pa.ipnets.ru/public/viewlog.php
```

**Comment by DavidGoodwin** [comment]
Those undefined variable messages shouldn't be bad enough to trigger a http 500 style error. I suspect something else is going on.

**Comment by Alex-JTI** [comment]
I see PHP fatal error. It should be causing 500 right?

[Fri Jan 22 13:06:39.643533 2021] [proxy_fcgi:error] [pid 13595:tid 140670782109440] [client 93.92.199.232:54210] AH01071: Got error 'PHP message: PHP Notice: Undefined index: theme_favicon in /home/ipnets/public_html/postfixadmin/model/PFASmarty.php on line 73PHP message: Config::read(): attempt to read undefined config option "database_socket", returning nullPHP message: Config::read(): attempt to read undefined config option "password_expiration", returning nullPHP message: PHP Fatal error: In bool: expected config password_expiration to be a string, but received a NULL in /home/ipnets/public_html/postfixadmin/model/Config.php on line 140', referer: https://pa.ipnets.ru/public/main.php

**Comment by DavidGoodwin** [comment]
Sorry, the poor formating on github hid the fatal bit from me!

`PHP message: PHP Fatal error: In bool: expected config password_expiration to be a string, but received a NULL in /home/ipnets/public_html/postfixadmin/model/Config.php on line 140', referer: https://pa.ipnets.ru/public/main.php`

Try adding 

`$CONF['password_expiration'] = false;`

do your config.local.php

**Comment by Alex-JTI** [comment]
Fine, that did the job, thanks! Now I see all the pages.

Do I need to do anything about other PHP non-fatal errors?
Especially I'm worried about this one:
`PHP message: Config::read(): attempt to read undefined config option "database_socket", returning null`

**Comment by DavidGoodwin** [comment]
try : `$CONF['database_socket'] = '';` ?

**Comment by Alex-JTI** [comment]
The issue gone by setting this $CONF['database_socket'] to a real sock file value.

The pane lseems to be working but here is the rest of error remaining in the log:

[Sat Jan 23 17:12:01.909565 2021] [proxy_fcgi:error] [pid 6673:tid 140565446371072] [client 93.92.199.232:45472] AH01071: Got error 'PHP message: PHP Notice: Undefined index: theme favicon in /home/ipnets/public_html/postfixadmin/model/PFASmarty.php on line 73PHP message: PHP Notice: Undefined index: show_vacation in /home/ipnets/public_html/postfixadmin/functions.inc.php on line 2179PHP message: PHP Notice: Undefined index: show_disabled in /home/ipnets/public_html/postfixadmin/functions.inc.php on line 2189PHP message: PHP Notice: Undefined index: show_vacation in /home/ipnets/public_html/postfixadmin/functions.inc.php on line 2179PHP message: PHP Notice: Undefined index: show_disabled in /home/ipnets/public_html/postfixadmin/functions.inc.php on line 2189PHP message: PHP Notice: Undefined index: show_vacation in /home/ipnets/public_html/postfixadmin/functions.inc.php on line 2179PHP message: PHP Notice: Undefined index: show_disabled in /home/ipnets/public_html/postfixadmin/functions.inc.php on line 2189PHP message: PHP Notice: Undefined index: show_vacation in /home/ipnets/public_html/postfixadmin/functions.inc.php on line 2179PHP message: PHP Notice: Undefined index: show_disabled in /home/ipnets/public_html/postfixadmin/functions.inc.php on line 2189PHP message: PHP Notice: Undefined index: show_vacation in /home/ipnets/public_html/postfixadmin/functions.inc.php on line 2179PHP message: PHP Notice: Undefined index: show_disabled in /home/ipnets/public_html/postfixadmin/functions.inc.php on line 2189PHP message: PHP Notice: Undefined index: show_vacation in /home/ipnets/public_html/postfixadmin/functions.inc.php on line 2179PHP message: PHP Notice: Undefined index: show_disabled in /home/ipnets/public_html/postfixadmin/functions.inc.php on line 2189PHP message: PHP Notice: Undefined index: show_vacation in /home/ipnets/public_html/postfixadmin/functions.inc.php on line 2179PHP message: PHP Notice: Undefined index: show_disabled in /home/ipnets/public_html/postfixadmin/functions.inc.php on line 2189PHP message: PHP Notice: Undefined index: show_vacation in /home/ipnets/public_html/postfixadmin/functions.inc.php on line 2179PHP message: PHP Notice: Undefined index: show_disabled in /home/ipnets/public_html/postfixadmin/functions.inc.php on line 2189', referer: https://pa.ipnets.ru/public/list.php?table=domain

As far as I understand these variables theme favicon, show_vacation and show_disabled are not defined.
If this is OK to have them unset, can you update the code please to have it checked whether they are defined with something like `if(isset($_POST['theme favicon'])`
otherwise please have them set in the config file as default.

**Comment by DavidGoodwin** [comment]
the above changeset should fix your issue(s) - it will go into 3.3.5 when that eventually gets released.
